python/operator/bot/operator_bot.py

```diff
@@ -131,6 +133,45 @@ def divulge_aliases(event):
                     )]
         return client.submit(commands)
 
+    @client.ledger_created(Chat.UserSettings)
+    def user_setting_change(event):
+        logging.info(f'On {Chat.UserSettings}')
+        archive_after = event.cdata['archiveMessagesAfter']
+        message = f"archive retention has been changed to `{archive_after}s`"
+        client.submit(exercise_by_key(
+            Chat.Chat,
+            {'_1': client.party,
+             '_2': event.cdata['user']},
+            'Chat_PostMessage',
+            {
+                'poster': client.party,
+                'message': message,
+                'postedAt': f"{int(time.time())}"
```

Comment:
Obviously changing `postedAt` would be an unrelated change to what you were trying to do, but I find it really confusing that `postedAt`, defined as `Text` in the models and frequently cast to `int` in the Python code, is actually a timestamp…

Reply:
Me too - dimitri had a reason for using Text when I asked him about it a while back, but I dont recall what it was :p

Reply:
I assume this is a static time / logical time independence hedge, no?

of course if a DAML timestamp were to be used here then there would be an implicit dependency on the ledger time model, whereas now a textual UTC specified datetime can be logically and operationally independently specified.

although now that I think on it some more, it's not immediately clear from the code whether this postedAt value specifies the message *sent* or *received* time as measured from the sending *client* or receiving *server*, so that might be good to disambiguate some.

Reply:
shall we call him while he's enjoying his favorite cheese?

Reply:
a'right, ima changing it to `int` and see if it breaks